apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -420,22 +420,25 @@ type RolloutStrategy struct {
 	// +optional
 	RollingUpdate *RollingUpdateConfig `json:"rollingUpdate,omitempty"`
 
-	// ApplyStrategy describes how to resolve the conflict if the resource to be placed already exists in the target cluster
-	// and is owned by other appliers.
+	// ApplyStrategy describes when and how to apply the selected resources to the target cluster.
 	// +optional
 	ApplyStrategy *ApplyStrategy `json:"applyStrategy,omitempty"`
 }
 
-// ApplyStrategy describes how to resolve the conflict if the resource to be placed already exists in the target cluster
-// and whether it's allowed to be co-owned by other non-fleet appliers.
+// ApplyStrategy describes when and how to apply the selected resource to the target cluster.
 // Note: If multiple CRPs try to place the same resource with different apply strategy, the later ones will fail with the
 // reason ApplyConflictBetweenPlacements.
 type ApplyStrategy struct {
 	// Type defines the type of strategy to use. Default to ClientSideApply.
-	// Server-side apply is a safer choice. Read more about the differences between server-side apply and client-side
-	// apply: https://kubernetes.io/docs/reference/using-api/server-side-apply/#comparison-with-client-side-apply.
+	// Server-side apply is more powerful and flexible than client-side apply.
+	// You SHOULD use server-side apply to safely resolve any potential drift between the
+	// original applied resource version and the current resource on the member cluster.
+	// Read more about the differences between server-side apply and client-side apply:
+	// https://kubernetes.io/docs/reference/using-api/server-side-apply/#comparison-with-client-side-apply.
+	// You can also use ReportDiff to only report the difference between the resource on the member cluster
+	// and the resource to be applied from the hub on all the selected clusters.
 	// +kubebuilder:default=ClientSideApply
-	// +kubebuilder:validation:Enum=ClientSideApply;ServerSideApply
+	// +kubebuilder:validation:Enum=ClientSideApply;ServerSideApply;ReportDiff
```

Comment:
Hi! My understanding is that this option is for apply diff, right? i.e., diff is found if a) a change has been made on the member cluster end w/o Fleet's acknowledgement (we need to track hashes of last successful apply results); and b) further apply op would lead to object changes. Shall we make this explicit in the comment?

Reply:
And this brings out the question of whether we'd like to support full diff as prev. explored in the UX doc, i.e., we also factor in fields that are not managed by Fleet -> I'll check with Infosys just to be certain.

Reply:
diff = diff between "the current version on the member cluster" and "the resource on the member cluster after the apply"